docs/naps/2-conda-based-packaging.md

```diff
@@ -0,0 +1,557 @@
+
+Adding the necessary functionality to the plugin dialog, including:
+
+* Adapting the current `Installer` class to support not only `pip` but also `conda/mamba` in a
+  single code base.
+* Adding support for `install`, `uninstall`, and `update` commands.
+* Populating the listing with data obtained from the napari hub API [^napari-hub-api].
+* Updating the listing to filter and mark plugins based on their availability on conda-forge.
+
+Some UX/UI improvements that are not part of this milestone but could be implemented in future
+releases include:
+
+* Making the plugin dialog a side panel / side dockwidget (similar to VS Code
+  [^vscode-extensions-ui])
+* Allow for simultaneous install of plugins. Currently multiple plugins can be selected for
+  install, uninstall and update, but each on of these actions, are queued and run sequentially.
+* Allow advanced users to install packages from PyPI or other local sources via drag-and-drop. This
+  presents many risks that could irreversibly disrupt the installation, though, so it needs to be
+  considered very carefully.
+
+### Milestone 4: Enabling in-app napari version updates
+
+Our first attempts [^in-app-update-pr] to implement this consisted of a pop-up dialog that would
+notify the user about the available updates. Accepting the dialog would trigger the creation of a
+new environment (e.g. `napari-0.4.16`) alongside the existing one (e.g. `napari-0.4.15`). While it
+could be argued that this is not an "update" per se, but a new installation, it fulfills the same
+goal: the new version is available. If the user wants to get rid of the old one, an option can
+be offered to auto-clean on success, or after not using it for some configurable time.
+
+However, having several versions of napari coexisting might risk questions like "which napari
+version is the default one?", "which one will be in charge of updating napari?", "what if the
+update logic changes over time?". There possible workarounds, but from the user perspective it could
+get confusing very fast.
+
+For that reason, we decided to slightly redesign our approach to in-app updates. We will still use
+several environments to keep things tidy and performant (see "Detailed description" for Milestone 2
+for more details), but the update checks will mimic something closer to a local server-client model
+or, more accurately, a `napari-updater` process (name subject to change; other options include
+`napari-launcher` or  `napari-manager`) that can be queried by a running `napari` instance.
+
+* `napari-updater`: We will create a separate standalone package to handle updates outside of the
+  `napari` process. This will be designed in an application agnostic way while targeting the needs
+  of the napari project. It will allow the user to handle updates but also manage the existing
+  napari installations (investigate issues, remove old ones, clean expired caches, etc).
```

Comment:
I love the concept of `napari-updater`! It can be a great place for us to recommend removing the older napari versions immediately upon getting a newer version and a place we can navigate OS specific edge-cases regarding installation/removing napari.